index.js

```diff
@@ -1,569 +0,0 @@
-            return this.startPc();
-        }
-    }
-}
-
-const downStairs = {
-    inventory: ['stairs', 'TV', 'bookcase', 'door', 'kitchen table'],
-    people: ['mom'],
-    desc: `Your mom sits at a table in the middle of the room. On the north wall is a TV, and in the northwest corner is a bookcase. \nIn the northeast corner are the stairs you came down, and to the south wall is the door.`,
-    mom: `MOM: Right. All boys leave home some day. It said so on TV.\nProf. Oak, next door, is looking for you.`,
-    bookcase: 'Crammed full of Pokemon Books!',
-    table: `The table has a flower in a vase sitting in the middle of it.`,
-    tv: 'There\'s a movie on tv. Four boys are walking on railroad tracks... I\'d better go, too!',
-    stairs: `These stairs lead to your bedroom!`,
-    door: `The entrance to your childhood home.`,
-}
-
-const laboratory = {
-    desc: `You walk into the laboratory where a woman and two aides mill about in the entrance in front of a row of eight bookcases. Beyond these shelves is the lab where your rival in standing. Behind him are more rows of shelves and two books on a desk next to another PC. On a table in the middle of the lab near your rival are three of those poke balls you've been hearing about.\nWhat do you want to do?`,
-    book: `It's encyclopedia-like, but the pages are blank!`,
-    pc: `There's an email message here!\n...\nCalling all Pokemon trainers!\nThe elite trainers of Pokemon league are ready to take on all comers! Bring your best Pokemon and see how you rate as a trainer! Pokemon League HQ Indigo Plateau\nPS: Prof. Oak! Please visit us!\n...`,
-    shelf: {
-        names: ['shelf', 'shelves', 'bookshelf', 'bookshelves', 'bookcase', 'bookcases', 'case', 'cases'],
-        desc: `Crammed full of Pokemon books!`,
-    },
-    pokeball: `Those are poke balls. They contain Pokemon!`,
-    notpokeball: `That's Prof. Oak's last pokemon!`,
-    inventory: ['pokeball1', 'pokeball2', 'pokeball3', 'book', 'pc', 'shelf', 'pokeball', 'pokeballs'],
-    people: [`rival`, 'woman', 'aide'],
-    rival: `Yo ${player.name[0]}!\nGramps isn't around!`,
-    woman: `"Prof Oak is the authority on Pokemon. Many pokemon trainers hold him in high regard."`,
-    aide: `"I study Pokemon as Prof. Oak's aide."`,
-//In this use case, "locked" means Oak is not yet in the lab and the player cannot choose a pokemon, or the player has already chosen a pokemon.
-    locked: true
-}
-
-let battleCommands = {
-    run: ['run', 'flee'],
-    fight: ['fight', 'attack'],
-    pokemon: ['pokemon'],
-    bag: ['items', 'bag', 'inventory']
-}
-
-let states = {
-    'bedRoom': { canChangeTo: ['downStairs'] },
-    'downStairs': { canChangeTo: ['palletTown', 'bedRoom'] },
-    'palletTown': { canChangeTo: ['laboratory', 'routeOne', 'downStairs'] },
-    'laboratory': { canChangeTo: ['palletTown'] },
-    'routeOne': { canChangeTo: ['palletTown'] }
-}
-
-let currentState = 'bedRoom';
```

Comment:
Nice implementation of state machines! Really makes your game logic flow nicely.  Leaving the state change to the `enterState` function does wonders for making your code more readable. Keep that separation of roles up! Very good practice.